pkg/db/db_sync_status.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	allCombs, err := h.DBReadUnsyncedAppsForTransfomerID(ctx, tx, id)
+	if err != nil {
+		return fmt.Errorf("DBBulkUpdateUnsyncedApps unable to read unsynced apps: %w", err)
+	}
+	if len(allCombs) == 0 {
+		logger.FromContext(ctx).Sugar().Warnf("Could not update all unsynced apps. Did not find any unsynced apps.")
+		return nil
+	}
+	now, err := h.DBReadTransactionTimestamp(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("DBBulkUpdateUnsyncedApps unable to get transaction timestamp: %w", err)
+	}
+
+	return h.executeBulkInsert(ctx, tx, allCombs, *now, id, status, BULK_INSERT_BATCH_SIZE)
+}
+
+func (h *DBHandler) DBRetrieveSyncStatus(ctx context.Context, tx *sql.Tx, appName, envName string) (*GitSyncData, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBRetrieveSyncStatus")
+	defer span.Finish()
+	if h == nil {
+		return nil, nil
+	}
+	if tx == nil {
+		return nil, fmt.Errorf("DBRetrieveSyncStatus: no transaction provided")
+	}
+
+	selectQuerry := h.AdaptQuery("SELECT transformerid, envName, appName, status FROM git_sync_status WHERE appName = ? AND envName= ? ORDER BY eslVersion DESC LIMIT 1;")
+	rows, err := tx.QueryContext(
+		ctx,
+		selectQuerry,
+		appName,
+		envName)
+	if err != nil {
+		return nil, fmt.Errorf("could not get git sync status for %q %q. Error: %w\n", envName, appName, err)
+	}
+
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			logger.FromContext(ctx).Sugar().Warnf("row closing error: %v", err)
+		}
+	}(rows)
+
+	var syncData GitSyncData
+	if rows.Next() {
+		err := rows.Scan(&syncData.TransformerID, &syncData.EnvName, &syncData.AppName, &syncData.SyncStatus)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				return nil, nil
+			}
+			return nil, fmt.Errorf("Error table scanning git_sync_status. Error: %w\n", err)
+		}
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return &syncData, nil
+}
+
+// These queries can get long. Because of this, we insert these values in batches
+func (h *DBHandler) executeBulkInsert(ctx context.Context, tx *sql.Tx, allEnvApps []EnvApp, now time.Time, id TransformerID, status SyncStatus, batchSize int) error {
+	queryPrefix := "INSERT INTO git_sync_status (created, transformerid, envName, appName, status) VALUES"
+	currentQuery := queryPrefix
+	if batchSize < 1 {
+		return fmt.Errorf("batch size needs to be a positive number")
+	}
+	for idx, currComb := range allEnvApps {
+		format := " ('%s', %d, '%s', '%s', %d)"
+		if idx%batchSize == 0 || idx == len(allEnvApps)-1 { // Is end of batch || tail
+			format += ";"
+			currentQuery += fmt.Sprintf(format, now.Format(time.RFC3339), id, currComb.EnvName, currComb.AppName, status)
+			_, err := tx.ExecContext(ctx, currentQuery)
+			if err != nil {
+				return err
+			}
+			currentQuery = queryPrefix
+		} else {
+			format += ","
+			currentQuery += fmt.Sprintf(format, now.Format(time.RFC3339), id, currComb.EnvName, currComb.AppName, status)
+		}
+	}
```

Comment:
I think this code would be more readible, if we just repeat the whole query.
So instead of
`insert into ... VALUES () () ();`
It would be 
`insert into ... VALUES (); insert into ... VALUES (); insert into ... VALUES ();`